ci/build_container/build_recipes/v8.sh

```diff
@@ -2,133 +2,55 @@
+# Patch wee8.
 
-# Build v8 static library.
-
-tools/dev/v8gen.py x64.release -- v8_monolithic=true v8_use_external_startup_data=false v8_enable_i18n_support=false v8_enable_gdbjit=false use_custom_libcxx=false
-ninja -v -C out.gn/x64.release v8_monolith
-
-# Install v8.
-
-mkdir -p "$THIRDPARTY_BUILD/include/v8/libplatform"
-cp -p include/v8*.h "$THIRDPARTY_BUILD/include/v8/"
-cp -p include/libplatform/*.h "$THIRDPARTY_BUILD/include/v8/libplatform/"
-cp -p out.gn/x64.release/obj/libv8_monolith.a "$THIRDPARTY_BUILD/lib/"
-
-cd ../..
-
-# Patch wasm-c-api.
-
-# 1. Disable hardcoded sanitizers.
+# 1. Fix handling of f64 globals.
+# 2. Force full GC when destroying VMs.
 
 cat <<\EOF | patch -p1
---- a/Makefile
-+++ b/Makefile
-@@ -8,9 +8,9 @@ V8_ARCH = x64
- V8_MODE = release
-
- WASM_FLAGS = -DDEBUG  # -DDEBUG_LOG
--C_FLAGS = ${WASM_FLAGS} -Wall -Werror -ggdb -O -fsanitize=address
-+C_FLAGS = ${WASM_FLAGS} -Wall -Werror -ggdb -O
- CC_FLAGS = ${C_FLAGS}
--LD_FLAGS = -fsanitize-memory-track-origins -fsanitize-memory-use-after-dtor
-+LD_FLAGS =
-
- C_COMP = clang
-
-EOF
-
-# 2. Don't leak implementation into headers.
-
-cat <<\EOF | patch -p1
---- a/include/wasm.hh
-+++ b/include/wasm.hh
-@@ -111,13 +111,8 @@ class vec {
-   size_t size_;
-   std::unique_ptr<T[]> data_;
-
+--- a/src/wasm/c-api.cc
++++ b/src/wasm/c-api.cc
+@@ -825,7 +825,7 @@ void global_set_f32(v8::Local<v8::Object> global, float val) {
+ void global_set_f64(v8::Local<v8::Object> global, double val) {
+   auto v8_object = v8::Utils::OpenHandle<v8::Object, i::JSReceiver>(global);
+   auto v8_global = i::Handle<i::WasmGlobalObject>::cast(v8_object);
+-  v8_global->SetF32(val);
++  v8_global->SetF64(val);
+ }
+
+ // Tables
+@@ -1107,7 +1107,7 @@ class StoreImpl {
+   StoreImpl() {}
+
+   ~StoreImpl() {
 -#ifdef DEBUG
-   void make_data();
-   void free_data();
--#else
--  void make_data() {}
--  void free_data() {}
--#endif
-
-   vec(size_t size) : vec(size, size ? new(std::nothrow) T[size] : nullptr) {
-     make_data();
---- a/src/wasm-v8.cc
-+++ b/src/wasm-v8.cc
-@@ -210,6 +210,14 @@ Stats stats;
-     if (data_) stats.free(Stats::STAT, data_.get(), Stats::VEC); \
-   }
-
-+#else
-+
-+#define DEFINE_VEC(type, STAT) \
-+  template<> void vec<type>::make_data() {} \
-+  template<> void vec<type>::free_data() {}
-+
-+#endif  // #ifdef DEBUG
-+
- DEFINE_VEC(byte_t, BYTE)
- DEFINE_VEC(Frame*, FRAME)
- DEFINE_VEC(ValType*, VALTYPE)
-@@ -231,8 +239,6 @@ DEFINE_VEC(Memory*, MEMORY)
- DEFINE_VEC(Extern*, EXTERN)
- DEFINE_VEC(Val, VAL)
-
--#endif  // #ifdef DEBUG
--
-
- ///////////////////////////////////////////////////////////////////////////////
- // Runtime Environment
-
++#if 1
+     reinterpret_cast<i::Isolate*>(isolate_)->heap()->PreciseCollectAllGarbage(
+         i::Heap::kNoGCFlags, i::GarbageCollectionReason::kTesting,
+         v8::kGCCallbackFlagForced);
 EOF
 
-# Build wasm-c-api.
-
-# TODO(PiotrSikora): respect CC/CXX/CFLAGS/CXXFLAGS/LDFLAGS upstream.
+# Build wee8 library.
```

Comment:
Could you comment on what is wee8 vis-a-vis v8 ?

Reply:
wee8 is supposed to be a trimmed down version of v8 that supports only WebAssembly (i.e. no JavaScript), but since this initiative just started, right now wee8 == v8.